notebooks/02-ingestion/13-native-transforms.ipynb

```diff
@@ -0,0 +1,875 @@
+    "            \"name\": \"referrer\",\n",
+    "            \"expression\": \"case_searched((\\\"referrer\\\" == 'unknown'),null,\\\"referrer\\\")\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "dataSchema = {\n",
+    "    \"dataSource\": table_name,\n",
+    "    \"timestampSpec\": dataSchema_timestampSpec,\n",
+    "    \"transformSpec\" : dataSchema_transformSpec,\n",
+    "    \"dimensionsSpec\": dataSchema_dimensionsSpec,\n",
+    "    \"granularitySpec\": dataSchema_granularitySpec\n",
+    "    }\n",
+    "\n",
+    "ingestion_spec = {\n",
+    "    \"type\": \"kafka\",\n",
+    "    \"spec\": {\n",
+    "        \"ioConfig\": ioConfig,\n",
+    "        \"tuningConfig\": tuningConfig,\n",
+    "        \"dataSchema\": dataSchema\n",
+    "    }\n",
+    "}\n",
+    "\n",
+    "print(json.dumps(ingestion_spec, indent=5))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "af961b1b-7b1b-40b0-b064-930a4b0b24c2",
+   "metadata": {},
+   "source": [
+    "Now submit the updated ingestion specification by running the cell below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "76a28ea7-f7a1-443b-9b41-90cfa713de02",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "requests.post(f\"{druid_host}/druid/indexer/v1/supervisor\", json.dumps(ingestion_spec), headers=druid_headers)\n",
+    "druid.sql.wait_until_ready(table_name, verify_load_status=False)\n",
+    "display.table(f'{table_name}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5bc99dae-f89c-4a23-9a12-54c95ff759f4",
+   "metadata": {},
+   "source": [
+    "Wait for a moment or two. This will allow Druid to apply the new configuration for the ingestion by shutting down the existing task, and starting a new ingestion task with the transforms.\n",
+    "\n",
+    "Run the following cell a number of times to see what effect this has had on the data in the table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0bc0e2e1-498c-49fc-af91-17b1935b63ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_now = datetime.now().strftime('%Y-%m-%dT%H:%M:%S')\n",
+    "\n",
+    "sql=f'''\n",
+    "SELECT\n",
+    "  \"__time\",\n",
+    "  \"keyword\",\n",
+    "  \"referrer\",\n",
+    "  \"product\"\n",
+    "FROM \"{table_name}\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'PT15S/{time_now}')\n",
+    "ORDER BY __time DESC\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "407a6da9-94b7-4db4-a022-a1e77676b1b9",
+   "metadata": {},
+   "source": [
+    "## Use a function to add a new column\n",
+    "\n",
+    "Since all the examples above have the same `name` as an existing column, the supplied `expression` results in the data in that column being overwritten. In this example, you will use a new column name in `name` to create a new dimension that uses data from incoming data.\n",
```

Comment:
```suggestion
    "Since all the examples above use the same `name` as an existing column, the supplied `expression` overwrites the data in that column. In this example, you use a new column name to create a new dimension from the incoming data.\n",
```